docs/assets/diagrams/transfer/02-create-transaction.puml

```diff
@@ -0,0 +1,132 @@
+@startuml
+
+title Create Transaction
+
+hide footbox
+
+autonumber
+
+box "Mobile Device" #Light
+  actor Alice
+  participant "PISP App" as App
+end box
+box "PISP" #Light
+  database Firebase
+  participant "PISP Server" as Server
+end box
+box "Mojaloop" #Light
+  participant "Third-party\nAPI Adapter" as Adapter
+end box
+
+Alice -> App: I want to make this transaction
+
+rnote right of App #Light 
+  ""createDoc(""
+  ""  "transactions/users/1234/<uuid>",""
+  ""  {""
+  ""    payee: {""
+  ""      phoneNumber: "+1-222-222-2222"""
+  ""    },""
+  ""    status: "PENDING"""
```

Comment:
Sorry, can I clarify what do you mean by "transaction notification"? are you referring to the result of party lookup?

> I think perhaps the server should be the only one permitted to update the status field, right?

I'm actually not sure about this. In the current design, after the user confirms the payee info (in diagram 3a), it will also change the status from `PENDING_PAYEE_CONFIRMATION` to `PAYEE_CONFIRMED`. 

However, if you want the server to be the only one who has write access to the Firebase, I actually think the app could talk directly to the server (instead of writing the message in Firebase which then notify the server). That way, the server will be the only one who always writes to the database and the app only needs to have a read permission to keep listening to all the changes.

This is actually also related to my concern when writing the sequence diagrams for linking. Because I don't think the app should let Firebase store signed challenge or auth token even for just a moment, and instead need to send its message directly to the server.